hls/decode-util_test.go

```diff
@@ -250,8 +249,8 @@ func TestReadMediaPlaylist(t *testing.T) {
 		if s.URI != newP.Segments[i].URI {
 			t.Errorf("Expected URI to be %s, but got %s", s.URI, newP.Segments[i].URI)
 		}
-		if s.Map != nil && !reflect.DeepEqual(s.Map, newP.Segments[i].Map) {
-			t.Errorf("Expected %d Segment Map tp be %v, but got %v", i, s.Map, newP.Segments[i].Map)
+		if s.Map != nil && s.Map.URI != newP.Segments[i].Map.URI {
```

Comment:
Definitely, thanks for that go-cmp link @bradleyfalzon . Led me to the idea to just create a Equal() function for the type. If we ever use the go-cmp for the check it'll use it as well.

Appreciate it!